### ISSUE TITLE
Niggles...

### ISSUE THREAD

**Issue by crisward** [amberframework/micrate#13]
Thanks again for the updates. I could be using it wrong, but there seem to be a couple of probably easy to rectify niggles. I don't mind doing a pull request,  but it probably wouldn't take you long to add these...

### Micrate.up / down

This line `target = all_migrations.keys.sort.last`
Assumes there are some migrations. If the directory is empty you get an index of of bounds error.

Something like this may be better

```
if !all_migrations.keys.empty?
  target = all_migrations.keys.sort.last
  migrate(all_migrations, current, target, db, migrations_table_suffix)
else
  puts "No migrations found in #{migrations_path}"
end
```

### File Matching

Line 140 is currently 

```
  private def self.migrations_by_version(migrations_path)
    Dir.entries(migrations_path)
       .select { |name| File.file? File.join(migrations_path, name) }
       .select { |name| /^\d+_.+\.sql$/ =~ name }
       .map { |name| Migration.from_file(migrations_path, name) }
       .index_by { |migration| migration.version }
  end
```
It could be better to make the regex less fussy. Users of micrate may decide to manually write their own migrations without using the cli tool. At the moment the regex requires the filename to start with number, then an underscore then anything, then `.sql`. All of this is fine, except the underscore. Perhaps change to `^\d+.+\.sql$`?

### Sql Errors?

Finally, sql errors seem to be getting swallowed. It would be nice if I migration fails to be able to get the output. Perhaps it does this and I've missed how to use it. Any feedback on this would be cool.

Again thanks for this, and if you'd prefer a pull request for these updates, let me know.

**Comment by juanedi** [comment]
Thanks for reporting these!

- Error handling for empty directories: totally agree we should improve these. I'm leaving some comments in your pull request.

- File name regex: not sure about this one. `Migration#from_file` assumes an underscore to separate the version from the migration name. We could change this too, but it doesn't change the fact that there is a structure for the migration file names. The only difference with the current one is that we would be allowing variable size for the version part. The version is suppose to correspond to a timestamp, so I don't see a reason to change this. I guess I prefer having the users write the underscore :-)

- SQL errors: seems reasonable! Would you mind opening a separate issue to track this one?

Thanks for contributing! I'm glad you found the tool useful.

**Comment by faustinoaq** [comment]
Hi @crisward Is this still an issue?

Now `micrate` is part of `amberframework` :tada:

**Comment by drujensen** [comment]
This is fixed in v0.9.0
